forms-flow-api/src/formsflow_api/services/form_process_mapper.py

```diff
@@ -1188,3 +1193,119 @@ def create_form_with_process(cls, request_data, is_designer):
+        AuthorizationService.create_or_update_resource_authorization(
+            authorization_data, is_designer=is_designer
+        )
+        return authorization_data
+
+    def handle_process_data(
+        self, process_data, **kwargs
+    ):  # pylint:disable=unused-argument
+        """Handler function for process updates."""
+        current_app.logger.debug("Updating process details..")
+        process_id = process_data.get("id")
+        if not process_id:
+            raise BusinessException(BusinessErrorCode.INVALID_INPUT)
+
+        process_payload = process_data.get("processData")
+        process_type = process_data.get("processType")
+        return ProcessService.update_process(process_id, process_payload, process_type)
+
+    def update_form_process(self, request_data, mapper_id, is_designer):
+        """Update form, process, authorizations and mapper details in one call.
+
+        Args:
+            request_data (dict): Combined update payload that may include any of:
+                - formData (dict): Formio form JSON; must include `_id` when provided.
+                - mapper (dict): Mapper fields to update (e.g., formName, status,
+                taskVariables, etc.). If `taskVariables` is present, it will be
+                persisted and filter variables will be updated accordingly.
+                - authorizations (dict): Authorization configuration; created/updated for the
+                associated resource when provided.
+                - process (dict): { id, processData, processType } used to update the process.
+            mapper_id (int): ID of the mapper record to update.
+            is_designer (bool): Whether the caller has designer permissions; used for
+                authorization updates.
+
+        Returns:
+            dict: A dictionary containing any of the updated sections, with keys among
+                { "formData", "mapper", "authorizations", "process" } depending on input.
+
+        Raises:
+            BusinessException: If required identifiers are missing or entities are invalid.
+        """
+        response = {}
+
+        # Registry of handler functions for different data types
+        handlers = {
+            "formData": self.handle_form_data,
+            "mapper": self.handle_mapper_data,
+            "authorizations": self.handle_authorization_data,
+            "process": self.handle_process_data,
+        }
+
+        # Common context for all handlers
+        handler_context = {"mapper_id": mapper_id, "is_designer": is_designer}
+
+        # Process each supported data type present in the request
+        # Note: Update order matters
+        # Updating process last avoids autoflush errors caused by transient str XML in
+        # the Process entity before it's converted to bytes.
+
+        for key in ("formData", "mapper", "authorizations", "process"):
+            data = request_data.get(key)
+            if data is not None:
```

Comment:
Will the missing keys in response cause any errors down the line? Should we add a default `empty / None` value or is this safe?